### PR TITLE
Don't chown/chmod volumes if not needed.

### PR DESCRIPTION
**Pull request by cpuguy83**
Doing a chown/chmod automatically can cause `EPERM` in some cases (e.g.
with an NFS mount). Currently Docker will always call chown+chmod on a
volume path unless `:nocopy` is passed in, but we don't need to make
these calls if the perms and ownership already match and potentially
avoid an unnecessary `EPERM`.

### CODE REVIEW

**Comment by justincormack** [comment]
Are you sure this makes a difference? Chown/chmod to same ID does not give
an EPERM error.

**Comment by cpuguy83** [comment]
Oh good point, I'll have to verify, was just going off user input here.

**Comment by cpuguy83** [comment]
Have an nfs export without `no_root_squash` set, and chowning to the same UID/GID gives `EPERM`

```
root@7e7dff800928:/# mount -t nfs localhost:/data /data2
root@7e7dff800928:/# chown root:root /data2
chown: changing ownership of '/data2': Operation not permitted
root@7e7dff800928:/# chown 0:0 /data2
chown: changing ownership of '/data2': Operation not permitted
```

```
drwxr-xr-x   2 root root 4.0K Jun 16 14:53 data
drwxr-xr-x   2 root root 4.0K Jun 16 14:53 data2
```

**Comment by justincormack** [comment]
Oh ok, that's kind of invalid but LGTM to work around it then, but let's
add a comment.

**Comment by cpuguy83** [comment]
Comment added.

File: container/container_unix.go [hunk 1]
```diff
@@ -441,11 +441,26 @@ func copyOwnership(source, destination string) error {
...
+	if err != nil {
 		return err
 	}
 
-	return os.Chmod(destination, os.FileMode(stat.Mode()))
+	// In some cases, even though UID/GID match and it would effectively be a no-op,
+	// this can return a permission denied error... for example if this is an NFS
+	// mount.
+	// Since it's not really an error that we can't chown to the same UID/GID, don't
+	// even both trying in such cases.
```
Comment (thaJeztah) [inline container/container_unix.go]: I assume you meant to type s/both/bother/

**Review by thaJeztah — APPROVED** [review]
One nit, but LGTM otherwise

**Comment by cpuguy83** [comment]
Comment fixed. All green + 2 LGTMs.
Merging.

**Pull request merged**